client/table.go

```diff
@@ -196,6 +200,134 @@ func (m *model) encodeColumnKey(primaryKey []byte, column string) []byte {
 	return append(key, column...)
 }
 
+// CreateTable creates a table from the specified schema. Table creation will
+// fail if the table name is already in use.
+func (db *DB) CreateTable(schema proto.TableSchema) error {
+	desc := proto.TableDescFromSchema(schema)
+	if err := proto.ValidateTableDesc(desc); err != nil {
+		return err
+	}
+
+	nameKey := keys.MakeNameMetadataKey(0, desc.Name)
+
+	// This isn't strictly necessary as the conditional put below will fail if
+	// the key already exists, but it seems good to avoid the table ID allocation
+	// in most cases when the table already exists.
+	if gr, err := db.Get(nameKey); err != nil {
+		return err
+	} else if gr.Exists() {
+		return fmt.Errorf("table \"%s\" already exists", desc.Name)
+	}
+
+	ir, err := db.Inc(keys.DescIDGenerator, 1)
+	if err != nil {
+		return err
+	}
+	desc.ID = uint32(ir.ValueInt() - 1)
+
+	return db.Txn(func(txn *Txn) error {
```

Comment:
This is going to have an incomprehensible error message. You might consider adding a TODO here to be cognizant of error messages when this is ported to server-side code.